kmod/uoa/uoa.md

```diff
@@ -80,6 +80,12 @@ Can we use private protocol, instead of IP option ?
 
 We may use a private `IPPROTO_XXX` protocol to encaplate UDP ? Looks like `IP+UOA+UDP`, simmilar with the idea of tunnel. But it may lead to firewall issues, note UDP/TCP are most popular, friendly to firewall or other middle box, but new protocol may not. And the new encaplation may have MTU issues for all packets. Another way is `IP+UDP+UOA+payload`, but it's difficult to identify the UOA and application payload.
 
+#### **Why support private protocol then ?**
+
+Currently, we have supported private protocol. It's because that we found not all l3-switch support IPv4 options, or even if support, there's speed limitation like 300pps. The reason from provider is the switch HW (chips) do not handle IP options, just have to drop the whole packet, or pass the packets with option to CPU for process, with a limited speed which is too poor to accept. On the other hand, the switch can "support" unkown IP protocol, we can forward this kind of packets.
```

Comment:
It's because that we found **that** not all **l3-switches** support IPv4 options, (如果意思是不是所有l3交换机支持IPv4选项的话，感觉后面这句"甚至不支持有点多余"而且这句话语法应该不能这么写。不知道你的意思是不是"不是支持所有IPv4选项") , **and** there's  

The reason from provider is **that** the switch **HWs** (chips) **doesn't** handle IP options **while they** just drop the whole packet or pass the packet with options to CPU for **processing at**  a limited speed which is too poor to accept. On the other hand, **if** the switch **could** "support" unkown IP protocols, **then** we can forward this kind of packets.